### PR TITLE
Generate API and controller, and add basic Travis CI

### PR DESCRIPTION
**Pull request by akrejcir**
Generated API and controller to watch SSP resource.

Added basic `.travis.yml`.

### CODE REVIEW

**Comment by akrejcir** [comment]
@omeryahud @ksimon1 @vatsalparekh

File: README.md [hunk 1]
```diff
@@ -1,2 +1,19 @@
 # ssp-operator
 Kubevirt SSP Operator
+
+### Building
+```shell script
+make operator-build
+```
+
+### Uploding to image repository
```
Comment (omeryahud) [inline README.md:9]: Maybe add
```
export IMAGE_REGISTRY=<registry>
export OPERATOR_IMAGE=<image_name>
export IMAGE_TAG=<image_tag>
```
To explicitly show how to modify the image location

Comment (omeryahud) [reply README.md:9]: Maybe add this to the `Building` section actually

File: Makefile [hunk 2]
```diff
@@ -16,4 +16,16 @@ operator-build: operator-sdk
 operator-push:
 	docker push $(IMAGE_REGISTRY)/$(OPERATOR_IMAGE):$(IMAGE_TAG)
 
-.PHONY: operator-build operator-push
+generate: operator-sdk
+	./operator-sdk generate k8s
+	./operator-sdk generate crds
```
Comment (omeryahud) [inline Makefile:21]: I think we should add `./operator-sdk generate csv` also

File: deploy/crds/ssp.kubevirt.io_v1_ssp_cr.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+apiVersion: ssp.kubevirt.io/v1
+kind: SSP
+metadata:
+  name: example-ssp
+spec:
+  # Add fields here
+  size: 3
```
Comment (omeryahud) [inline deploy/crds/ssp.kubevirt.io_v1_ssp_cr.yaml:7]: Please remove spec fields and use `spec: {}` instead

File: pkg/controller/ssp/ssp_controller.go [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+	// Watch for changes to primary resource SSP
+	err = c.Watch(&source.Kind{Type: &sspv1.SSP{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	// TODO(user): Modify this to be the types you create that are owned by the primary resource
+	// Watch for changes to secondary resource Pods and requeue the owner SSP
+	err = c.Watch(&source.Kind{Type: &corev1.Pod{}}, &handler.EnqueueRequestForOwner{
```
Comment (omeryahud) [inline pkg/controller/ssp/ssp_controller.go:54]: This is part of the example controller, we do not need this as we are not managing pods directly.

Comment (akrejcir) [reply pkg/controller/ssp/ssp_controller.go]: Done

File: deploy/crds/ssp.kubevirt.io_v1_ssp_cr.yaml (hunk 3 above)
Comment (akrejcir) [reply deploy/crds/ssp.kubevirt.io_v1_ssp_cr.yaml]: Done

File: Makefile (hunk 2 above)
Comment (akrejcir) [reply Makefile:21]: `operator-sdk generate csv` requires some parameters such as operator name, description, provider, maintainers' names and emails, etc.. I think we should add it in a new PR.

File: README.md (hunk 1 above)
Comment (akrejcir) [reply README.md]: Done

**Comment by akrejcir** [comment]
@omeryahud , can you enable Travis for this repo?

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+dist: xenial
+language: go
+go: "1.13"
```
Comment (vatsalparekh) [inline .travis.yml:3]: I see that my PR had this version in `go.mod` probably added by IDE, but I'm not sure if we want to pin go version
Also, not sure if we have to pin go version to 1.12 like kubevirt has done for downstream purpose
Thoughts? @omeryahud @ksimon1

File: Makefile (hunk 2 above)
Comment (omeryahud) [reply Makefile:21]: Sounds good, thanks

File: .travis.yml (hunk 5 above)
Comment (omeryahud) [reply .travis.yml:3]: I think we have a go-1.13 image downstream, I will check again

**Comment by omeryahud** [comment]
> @omeryahud , can you enable Travis for this repo?

Done, according to documentation travis should be working once the `.travis.yml` file is merged.

File: deploy/crds/ssp.kubevirt.io_ssps_crd.yaml [hunk 6]
```diff
@@ -0,0 +1,41 @@
+apiVersion: apiextensions.k8s.io/v1
```
Comment (ksimon1) [inline deploy/crds/ssp.kubevirt.io_ssps_crd.yaml:1]: Is the name of this crd correct?

Comment (akrejcir) [reply deploy/crds/ssp.kubevirt.io_ssps_crd.yaml:1]: The name was generated from the 'kind' and 'group', but we can change the name. Or change the 'kind' too, I was not sure what would be a good kind.

Comment (omeryahud) [reply deploy/crds/ssp.kubevirt.io_ssps_crd.yaml:1]: Yes, the name is correct

File: .travis.yml (hunk 5 above)
Comment (omeryahud) [reply .travis.yml:3]: Yes, there is a go-1.13 image downstream, this should be fine

File: deploy/crds/ssp.kubevirt.io_v1_ssp_cr.yaml [hunk 7]
```diff
@@ -0,0 +1,5 @@
+apiVersion: ssp.kubevirt.io/v1
+kind: SSP
+metadata:
+  name: example-ssp
```
Comment (vatsalparekh) [inline deploy/crds/ssp.kubevirt.io_v1_ssp_cr.yaml:4]: shouldn't this be changed to something other then example-ssp?

File: deploy/crds/ssp.kubevirt.io_v1_ssp_cr.yaml [hunk 8]
```diff
@@ -0,0 +1,5 @@
...
+spec: {}
```
Comment (vatsalparekh) [inline deploy/crds/ssp.kubevirt.io_v1_ssp_cr.yaml:5]: Can you add new line at the bottom?

File: deploy/crds/ssp.kubevirt.io_v1_ssp_cr.yaml (hunk 7 above)
Comment (akrejcir) [reply deploy/crds/ssp.kubevirt.io_v1_ssp_cr.yaml]: Changed to `kubevirt-ssp`.

File: deploy/crds/ssp.kubevirt.io_v1_ssp_cr.yaml (hunk 8 above)
Comment (akrejcir) [reply deploy/crds/ssp.kubevirt.io_v1_ssp_cr.yaml]: Added. I forgot to configure IDE to add them.

**Pull request merged**